trace/es/elasticsearch.go

```diff
@@ -0,0 +1,185 @@
+package es
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"runtime"
+	"strings"
+
+	"github.com/beatlabs/patron/trace"
+	"github.com/elastic/go-elasticsearch/v8"
+	"github.com/elastic/go-elasticsearch/v8/esapi"
+	"github.com/elastic/go-elasticsearch/v8/estransport"
+	"github.com/opentracing/opentracing-go"
+)
+
+const (
+	// ES Env Vars
+	DefaultHostEnv = "PATRON_ES_DEFAULT_HOST"
+	DefaultPortEnv = "PATRON_ES_DEFAULT_PORT"
+
+	// If no addresses are passed, hit localhost by default
+	DefaultHost = "http://localhost"
+
+	// If no ports are passed, hit default ES port by ...default
+	DefaultPort = "9200"
+
+	// span tag that denotes the host that responded to a query
+	respondentTag = "respondent"
+)
+
+type tracingInfo struct {
+	user  string
+	hosts []string
+}
+
+func (t *tracingInfo) startSpan(opName string, req *http.Request) (opentracing.Span, error) {
+
+	if req == nil {
+		return nil, fmt.Errorf("request is empty")
+	}
+
+	uri := req.URL.RequestURI()
+	method := req.Method
+
+	var bodyFmt string
+	if req.Body != nil {
+		if rawBody, err := ioutil.ReadAll(req.Body); err == nil {
+			req.Body = ioutil.NopCloser(bytes.NewReader(rawBody))
+			bodyFmt = string(rawBody)
+		}
+	}
+
+	return trace.EsSpan(req.Context(), opName, "go-elasticsearch", t.user, uri, method, bodyFmt, t.hosts), nil
+}
+
+func endSpan(sp opentracing.Span, rsp *http.Response) {
+	// Fetch the host that responded and inject it to the span
+	sp.SetTag(respondentTag, rsp.Request.URL.Host)
+
+	trace.FinishHTTPSpan(sp, rsp.StatusCode)
+}
+
+// transportClient is ES transport struct wrapped with tracing info
+type transportClient struct {
+	client *estransport.Client
+	tracingInfo
+}
+
```

Comment:
Im adding a comment to be consistent with go coding style, though I need to clarify that Perform here isn't actually meant to be exported (the end user should use the client for most cases) but needs to be so that the Transport interface can be implemented.